File: examples/php/linked_list_1.php
<?php
class Node {
    public $val;
    public $next;
    function __construct($val) { $this->val = $val; $this->next = null; }
}
class LinkedList {
    public $head;
    function append($val) {
        $node = new Node($val);
        if ($this->head === null) { $this->head = $node; return; }
        $cur = $this->head;
        while ($cur->next !== null) $cur = $cur->next;
        $cur->next = $node;
    }
}
$list = new LinkedList();
$list->append(1);
$list->append(2);
$list->append(3);
$cur = $list->head;
while ($cur !== null) { echo $cur->val . "\n"; $cur = $cur->next; }

File: examples/rust/linked_list_1.rs
#[derive(Debug)]
struct Node {
    val: i32,
    next: Option<Box<Node>>,
}

fn append(head: &mut Option<Box<Node>>, val: i32) {
    let mut cur = head;
    while cur.as_ref().unwrap().next.is_some() {
        cur = &mut cur.as_mut().unwrap().next;
    }
    cur.as_mut().unwrap().next = Some(Box::new(Node { val, next: None }));
}

fn main() {
    let mut head = Some(Box::new(Node { val: 1, next: None }));
    append(&mut head, 2);
    append(&mut head, 3);
    let mut cur = &head;
    while let Some(n) = cur {
        println!("{}", n.val);
        cur = &n.next;
    }
}
